timm/models/efficientvit_mit.py

```diff
@@ -0,0 +1,662 @@
+
+        # final projecttion
+        out = torch.transpose(out, -1, -2)
+        out = torch.reshape(out, (B, -1, H, W))
+        out = self.proj(out)
+
+        return out
+
+
+class EfficientViTBlock(nn.Module):
+    def __init__(
+        self,
+        in_channels,
+        heads_ratio=1.0,
+        dim=32,
+        expand_ratio=4,
+        norm=nn.BatchNorm2d,
+        act_func=nn.Hardswish,
+    ):
+        super(EfficientViTBlock, self).__init__()
+        self.context_module = ResidualBlock(
+            LiteMSA(
+                in_channels=in_channels,
+                out_channels=in_channels,
+                heads_ratio=heads_ratio,
+                dim=dim,
+                norm=(None, norm),
+            ),
+            nn.Identity(),
+        )
+        local_module = MBConv(
+            in_channels=in_channels,
+            out_channels=in_channels,
+            expand_ratio=expand_ratio,
+            use_bias=(True, True, False),
+            norm=(None, None, norm),
+            act_func=(act_func, act_func, None),
+        )
+        self.local_module = ResidualBlock(local_module, nn.Identity())
+
+    def forward(self, x):
+        x = self.context_module(x)
+        x = self.local_module(x)
+        return x
+
+
+class ResidualBlock(nn.Module):
+    def __init__(
+        self,
+        main: nn.Module or None,
+        shortcut: nn.Module or None,
+        post_act=None,
+        pre_norm: nn.Module or None = None,
+    ):
+        super(ResidualBlock, self).__init__()
+
+        self.pre_norm = pre_norm
+        self.main = main
+        self.shortcut = shortcut
+        self.post_act = post_act(inplace=True) if post_act else nn.Identity()
+
+    def forward_main(self, x):
+        if self.pre_norm is None:
+            return self.main(x)
+        else:
+            return self.main(self.pre_norm(x))
+
+    def forward(self, x):
+        if self.main is None:
+            res = x
+        elif self.shortcut is None:
+            res = self.forward_main(x)
+        else:
+            res = self.forward_main(x) + self.shortcut(x)
+            if self.post_act:
+                res = self.post_act(res)
+        return res
+
+
+class ClsHead(nn.Module):
+    def __init__(
+        self,
+        in_channels,
+        width_list,
+        n_classes=1000,
+        dropout=0,
+        norm=nn.BatchNorm2d,
+        act_func=nn.Hardswish,
+        global_pool='avg',
+    ):
+        super(ClsHead, self).__init__()
+        self.ops = nn.Sequential(
+            ConvNormAct(in_channels, width_list[0], 1, norm=norm, act_func=act_func),
+            SelectAdaptivePool2d(pool_type=global_pool, flatten=True, input_fmt='NCHW'),
+            nn.Linear(width_list[0], width_list[1], bias=False),
+            nn.LayerNorm(width_list[1]),
+            act_func(inplace=True),
+            nn.Dropout(dropout, inplace=False) if dropout else nn.Identity(),
+            nn.Linear(width_list[1], n_classes, bias=True),
+        )
```

Comment:
can this be flattened into the ClsHead (maybe rename ClassifierHead) module? I'd rather have something like `head.classifier` / `head.fc` instead of `head.ops.6` ... to be closer to https://github.com/huggingface/pytorch-image-models/blob/main/timm/layers/classifier.py